app/components/page-components/Chatbot/ChatbotTimers.vue.ts

```diff
@@ -15,16 +15,29 @@ export default class ChatbotTimers extends ChatbotBase {
     return this.chatbotApiService.state.timersResponse.pagination.current;
   }
 
+  get totalPages() {
+    return this.chatbotApiService.state.timersResponse.pagination.total;
+  }
+
   mounted() {
     // get list of timers
-    this.chatbotApiService.fetchTimers(this.currentPage);
+    this.fetchTimers(1);
+  }
+
+  get searchQuery() {
+    return this.query;
+  }
+
+  set searchQuery(value: string) {
+    this.query = value;
+    window.clearTimeout(this.timeOutFetchQuery);
+    this.timeOutFetchQuery = window.setTimeout(() => {
+      this.fetchTimers(this.currentPage, value);
+    }, 1000);
```

Comment:
Do you want to call `fetchTimers` when `this.query` is changed? If so use `debounce` decorator from lodash-decorators
```
query: string;

@Watch('query')
@Debounce(1000)
onQueryChangeHandler() {
   this.fetchTimers(this.currentPage, this.query);
}
```